File: client/src/dto/MatchInfoDto.ts
import { CellInfoDto } from "./CellInfoDto";
import { PartialPlayerInfoDto } from "./PlayerInfoDto";

// TODO : remove the player info dtos client side
export interface MatchInfoDto {
    id: string;
    roomId: string;
    boardArray: CellInfoDto[][];
    player1?: PartialPlayerInfoDto;
    player2?: PartialPlayerInfoDto;
    currentTurn: number;
}

File: server/dto/match_info_dto.py
from dataclasses import dataclass

from dto.base_dto import BaseDto
from dto.cell_info_dto import CellInfoDto
from dto.player_info_dto import PlayerInfoDto


@dataclass
class MatchInfoDto(BaseDto):
    id: str
    boardArray: list[list[CellInfoDto]]
    player1: PlayerInfoDto
    player2: PlayerInfoDto
    currentTurn: int
    started: bool
